docs/changehistory/NextVersion.md

```diff
@@ -62,7 +62,11 @@ To disable external textures, pass a `TileAdmin` to [IModelApp.startup]($fronten
   IModelApp.startup({ tileAdmin });
 ```
 
-Disabling this feature will incur a performance penalty. The option to disable this feature will likely be removed in the future.
+Disabling this feature is not recommended and will incur a performance penalty. The option to disable the feature will likely be removed in the future.
+
+## ECClass Ids in changed events
+
+The events generated by [TxnManager]($backend) and [BriefcaseTxns]($frontend) to notify of changes produced by [Txns]($docs/learning/InteractiveEditing.md) now include - in addition to the Ids of the changed [Element]($backend)s or [Model]($backend)s - the Id of each changed entity's [ECClass]($docs/bis/intro/glossary/#ecclass). The [OrderedId64Array]($bentleyjs-core) properties of [TxnChangedEntities]($backend) are deprecated in favor of the new [EntityIdAndClassIdIterable]($common) properties. Iterating the contents of a [ChangedEntities]($common) is easy using [ChangedEntitiesIterable]($common).
```

Comment:
To my knowledge, this will be the first time we use ECClassId in our public api. Should you give an example of how the classId can be used (does it involve converting to class name, which I believe is only @internal?)

Reply:
I wanted to but (1) ECClass itself is `@beta` (so I linked to the .md instead) and (2) the only remotely straightforward way I could find to go from classId to class name was via `IModelDb.nativeDb` which is `@internal`.
I found it odd that `EntityMetadata` does not include the class Id.
I think the metadata APIs could use some straightening out. e.g., how do I find out if my class Id is-a SpatialElement, for example?

Reply:
@khanaffan are there really no other public APIs where we use class Ids? Are there some examples or documentation?

Reply:
we've been trying to avoid ECClassId due to the iModel-specific-ness of it. For example, you can use the `ClassRegistry` on the backend to get a `typeof Entity`  from className, and then from that do `instance of SpatialElement` test.

But i suppose this might be a good time to add public `classIdToName` and `classNameToId` methods on `IModelDb` if we're going to start returning classIds.

Reply:
I suppose you could return className instead of classId in your iterable?

Reply:
your indexed array would have the (unique) class names, right?

Reply:
send that first, then the batches

Reply:
> send that first, then the batches

Then event listeners need to be stateful, storing the list of class names and keeping track of when the last batch is received.
On frontend, class name gets you very little more than class Id - you still can't do is-a checks without async backend calls.
Shall I just not include class information in the frontend events?

Reply:
That would be my vote for now. Lets wait until there's a valid use case.

Reply:
We support both className and classId in ECSQL api. But as keith mention emphasis on using className and all imodel.js api use classNames. If you construct className with schema-alias it should be smaller because schema name are much larger. So instead of `BisCore:Element` you can have `bis:Element`.  

If you have to perform alot of is-a check we can bring  filtered ec_baseClass table content to frontend It should not be more then few kbytes as for even very large schema set i have seen them around 7k rows and of type int. Let me know if need that we can save alot of asyn call to backend.

Reply:
Changed to omit class Ids in frontend events. EC metadata needs to be made more publicly accessible as part of a separate PR. For example, MetaDataRegistry is marked `@internal`.